lib/modules/datasource/hermit/index.ts

```diff
@@ -0,0 +1,165 @@
+ */
+export class HermitDatasource extends Datasource {
+  static readonly id = 'hermit';
+
+  override readonly customRegistrySupport = true;
+
+  override readonly registryStrategy = 'first';
+
+  override readonly defaultVersioning = id;
+
+  override readonly defaultRegistryUrls = [
+    'https://github.com/cashapp/hermit-packages',
+  ];
+
+  pathRegex: RegExp;
+
+  constructor() {
+    super(HermitDatasource.id);
+    this.http = new GithubHttp(id);
+    this.pathRegex = regEx('^\\/(?<owner>[^/]+)\\/(?<repo>[^/]+)$');
+  }
+
+  @cache({
+    namespace: `datasource-hermit-package`,
+    key: ({ registryUrl, packageName }: GetReleasesConfig) =>
+      `${registryUrl ?? ''}-${packageName}`,
+  })
+  async getReleases({
+    packageName,
+    registryUrl,
+  }: GetReleasesConfig): Promise<ReleaseResult | null> {
+    logger.trace(`HermitDataSource.getReleases()`);
+
+    if (!registryUrl) {
+      logger.error('registryUrl must be supplied');
+      return null;
+    }
+
+    const parsedUrl = parseUrl(registryUrl);
+    if (parsedUrl === null) {
+      logger.warn({ registryUrl }, 'invalid registryUrl given');
+      return null;
+    }
+
+    if (!registryUrl.startsWith('https://github.com/')) {
+      logger.warn({ registryUrl }, 'Only Github registryUrl is supported');
+      return null;
+    }
+
+    const items = await this.getHermitSearchManifest({
+      packageName,
+      registryUrl,
+    });
+
+    if (items === null) {
+      return null;
+    }
+
+    const res = items.find((i) => i.Name === packageName);
+
+    if (!res) {
+      logger.debug({ packageName, registryUrl }, 'cannot find hermit package');
+      return null;
+    }
+
+    const sourceUrl = res.Repository;
+
+    return {
+      sourceUrl,
+      releases: [
+        ...res.Versions.map((v) => ({
+          version: v,
+          sourceUrl,
+        })),
+        ...res.Channels.map((v) => ({
+          version: v,
+          sourceUrl,
+        })),
+      ],
+    };
+  }
+
+  /**
+   * getHermitSearchManifest fetch the index.json from release
+   * named index, parses it and returned the parsed JSON result
+   */
+  @cache({
+    namespace: `datasource-hermit-search-manifest`,
+    key: ({ registryUrl }: GetReleasesConfig) => registryUrl ?? '',
+  })
+  async getHermitSearchManifest({
+    registryUrl,
+  }: GetReleasesConfig): Promise<HermitSearchResult[] | null> {
+    const u = parseUrl(registryUrl);
+    const host = u?.host ?? '';
+    const groups = this.pathRegex.exec(u?.pathname ?? '')?.groups;
+
+    if (!groups) {
+      return null;
+    }
+
+    const { owner, repo } = groups;
+
+    const apiBaseUrl = getApiBaseUrl(`https://${host}`);
+
+    const indexRelease = await this.http.getJson<GithubRelease>(
+      `${apiBaseUrl}repos/${owner}/${repo}/releases/tags/index`
+    );
+
+    // finds asset with name index.json
+    const asset = indexRelease.body.assets.find(
+      (asset: GithubReleaseAsset) => asset.name === 'index.json'
```

Comment:
```suggestion
      (asset) => asset.name === 'index.json'
```
typescript should be able to resolve the type